### DISCUSSION TITLE
Sneak Peak! Wasm

### DISCUSSION THREAD

**Discussion by danwalmsley** [AvaloniaUI/Avalonia#6911] [General]
https://user-images.githubusercontent.com/4672627/141182673-127f463b-e953-4c15-b423-c6babc100d58.mp4

This is very early work! However Avalonia ontop of Blazor and Wasm is NOW viable and coming soon!

This particular example is now hosted at https://blaze.avaloniaui.net/

**Comment by aboimpinto** [comment]
very good ... can't wait for the moment to put my hands on it!!!

**Comment by npolyak** [comment]
Great job guys! Looking forward to Avalonia conquering the world!

**Comment by kekekeks** [comment]
https://blaze.avaloniaui.net/

**Reply by npolyak** [reply]
Pretty amazing!!!

**Comment by timothyparez** [comment]
This is awesome!

**Comment by robloo** [comment]
Well that is certainly unexpected! Makes sense to go the Blazor route where AOT is supported now and Skia support is pretty good (not to mention C#). That means performance is likely tolerable too (seems good in the video).

What about integration with 3rd party controls for Blazor or even JavaScript/TypeScript? Is there any support for embedding HTML content or access to the DOM? I imagine integration is almost non-existent as Avalonia is just rendering itself in Skia still.

**Reply by jmacato** [reply]
no it can't... not yet atleast :)

**Reply by maxkatz6** [reply]
Need to remember that we render (and will render) to a canvas. So it will be possible to put html component on top of canvas, but not inside of it between avalonia components.

**Reply by robloo** [reply]
Thanks, this is as expected then. By rendering to a canvas it should also be possible to make an Avalonia app a subview of an HTML page as well (not a full screen app).

**Reply by maxkatz6** [reply]
Yes. Not right now, but it was planned, as well as multiple canvases (==multiple views/windows) at the same page.

**Reply by npolyak** [reply]
Great. If this is implemented, Avalonia will essentially beat Silverlight at its peak in browser functionality.

**Reply by schnerring** [reply]
> Yes. Not right now, but it was planned, as well as multiple canvases (==multiple views/windows) at the same page.

This would be so amazing! Can I track the status of multi-canvas somewhere? I've looked through the issues and at the roadmap but couldn't find anything.

**Comment by timunie** [comment]
Is there a newer version of the preview available ?

**Reply by timunie** [reply]
I have heard about this in telegram, there is no newer version online yet, but it is worked on. So let's be patient :-)

**Reply by kekekeks** [reply]
The wasm backend is included in `0.10.11`, you can try it using `avalonia.xplat` dotnet template.

**Reply by timunie** [reply]
Well, I was talking about the online version of the control catalog. Sorry to be not clear enough in the question. Thank you for the answer anyway and have a great weekend 👍
